sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/GroupingState.java

```diff
@@ -20,34 +20,23 @@
 import org.apache.beam.sdk.transforms.Combine.CombineFn;
 
 /**
- * State for a single value that is managed by a {@link CombineFn}. This is an internal extension
- * to {@link CombiningState} that includes the {@code AccumT} type.
+ * State that combines multiple {@code InputT} values using a {@link CombineFn} to produce a single
+ * {@code OutputT} value.
  *
  * @param <InputT> the type of values added to the state
- * @param <AccumT> the type of accumulator
  * @param <OutputT> the type of value extracted from the state
  */
-public interface AccumulatorCombiningState<InputT, AccumT, OutputT>
-    extends CombiningState<InputT, OutputT> {
-
-  /**
-   * Read the merged accumulator for this combining value. It is implied that reading the
-   * state involes reading the accumulator, so {@link #readLater} is sufficient to prefetch for
-   * this.
-   */
-  AccumT getAccum();
-
+public interface GroupingState<InputT, OutputT> extends ReadableState<OutputT>, State {
```

Comment:
Since it is a shared superclass right now, to move it out I'll need to make it a wrapper instead. It will be easy, but will be actual code, so I'd defer for a follow-up. (And the wrapper can be made first, and ported to separately, so we don't need a synchronous worker change)